src/transformers/benchmark/benchmark.py

```diff
@@ -87,8 +87,18 @@ def _prepare_inference_func(self, model_name: str, batch_size: int, sequence_len
 
         if self.args.torchscript:
             config.torchscript = True
-        if self.args.with_lm_head:
-            model = MODEL_WITH_LM_HEAD_MAPPING[config.__class__](config)
+
+        has_model_class_in_config = hasattr(config, "architecture") and len(config.architectures) > 1
+        if not self.args.only_pretrain_model and has_model_class_in_config:
+            try:
+                model_class = config.architectures[0]
```

Comment:
Tagging @LysandreJik @julien-c for notification. Benchmarks are extended to use the model class as written in their `config.architecture` list. 

Merging for now, I think this should be good.